### PR TITLE
Kathirsvn/readme update

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:
1. README Cleanup
2. Created a separate dev guide read me
3. Fixed the .env.example that will be prepared and used for running tests.
**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Changes manually tested
- [ ] Automated Tests added/updated
- [X] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by kathirsvn** [comment]
There are descriptions to be added for the operations (which I marked as '_Description to be added_'), which I'm working on in parallel. Please skip the descriptions of the operations during the review for now, ll update once I complete adding that as well.

File: README.md [hunk 1]
```diff
@@ -1,251 +1,220 @@
-# stargate-mongoose
+# stargate-mongoose ![ci-tests](https://github.com/stargate/stargate-mongoose/actions/workflows/ci-tests.yml/badge.svg)
 
-![ci-tests](https://github.com/stargate/stargate-mongoose/actions/workflows/ci-tests.yml/badge.svg)
-
-`stargate-mongoose` is a mongoose driver for [JSON API Server](https://github.com/stargate/jsonapi)/[Astra DB](https://astra.datastax.com).
-
-## Table of contents
+`stargate-mongoose` is a mongoose driver for [JSON API Server](https://github.com/stargate/jsonapi) which runs on top of Apache Cassandra / DataStax Enterprise.
```
Comment (jeffreyscarpenter) [inline README.md]: nit: I don't like calling it a "Server". I'd just say JSON API.

File: README.md [hunk 2]
```diff
@@ -1,251 +1,220 @@
...
 
 1. [Quickstart](#quickstart)
-2. [Testing](#testing)
-3. [Compatability](#compatability)
+2. [Architecture](#architecture)
+3. [Features](#compatability)
```
Comment (jeffreyscarpenter) [inline README.md]: spelling: compatibility

Also, shouldn't the anchor should be renamed features anyway?

Or, I guess now we have Versions Compatibility and Features. Both should be in the ToC, right?

File: README.md [hunk 3]
```diff
@@ -1,251 +1,220 @@
...
-npm i -s stargate-mongoose
-```
-
-### Connect to an Astra DB/ JSON API local instance
-```javascript
-import mongoose from 'mongoose';
-import { driver, collections } from 'stargate-mongoose';
+Prerequisites:
+npm, node, Docker (for testing the sample app locally using docker compose)
+- Start `Docker` on your local.
```
Comment (jeffreyscarpenter) [inline README.md]: local machine

File: README.md [hunk 4]
```diff
@@ -1,251 +1,220 @@
...
+### High level architecture
+<img src="stargate-mongoose-arch.png" alt="stargate-mongoose usage end to end architecture"/>
+
+### Components
+- Cassandra Cluster - Apache Cassandra / DataStax Enterprise Cluster as backend database.
+- Stargate Coordinator Nodes - [Stargate](https://stargate.io/) is an opensource Data API Gateway for Cassandra. Coordinator is one of the primary components of Stargate which connects the API layer to the backend database. More details can be found [here](https://stargate.io/docs/latest/concepts/concepts.html#stargate-v2-0).
+- Stargate JSON API - [JSON API Server](https://github.com/stargate/jsonapi) is an opensource JSON API that runs on top of Stargate's coordinator.
+- JavaScript Clients that use mongoose - mongoose is an elegant mongodb object modeling library for node.js applications. By implementing a driver required by the mongoose interface to connect to the JSON API server instead of native mongodb access layer, now a JavaScript client can store/retrieve documents on an Apache Cassandra/DSE backend.
+
+## Version compatibility
```
Comment (jeffreyscarpenter) [inline README.md:96]: Should we mention that it has to be a DSE backend right now? I guess that's an indirect dependency though.

File: DEVGUIDE.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+3. [Contributing](CONTRIBUTING.md)
+4. [Creating a release](#releasing)
+
+## Build & Test
+
+Prerequisites:
+- [Docker](https://docker.com/) / [JSON API Server](https://github.com/stargate/jsonapi)
+
+### Build
+- Run `npm install && npm run build`
```
Comment (jeffreyscarpenter) [inline DEVGUIDE.md]: Maybe this is overdoing it, but if we make all these commands in code blocks then there will be a copy button, right?

File: DEVGUIDE.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Run below to genreate API documentation. This takes the `APIReference.hbs` and the library code as input and generates APIReference.md file.
+```shell
+npm run build:docs
+```
+
+## Releasing
+
+We are using [npm-publish](https://github.com/JS-DevTools/npm-publish) to handle publishing, so to publish a release to NPM, we just need to adjust the version in `package.json`.
+
+In order to do so, form up a PR with just the version change in it with the output from the changelog in the body of the PR (generate it from the releases ui of github).
```
Comment (jeffreyscarpenter) [inline DEVGUIDE.md]: nit: ui -> UI

File: README.md [hunk 7]
```diff
@@ -1,251 +1,220 @@
...
+    - http://localhost:8097/addproduct
+    - http://localhost:8097/getproducts
+
+## Architecture
+### High level architecture
+<img src="stargate-mongoose-arch.png" alt="stargate-mongoose usage end to end architecture"/>
+
+### Components
+- Cassandra Cluster - Apache Cassandra / DataStax Enterprise Cluster as backend database.
+- Stargate Coordinator Nodes - [Stargate](https://stargate.io/) is an opensource Data API Gateway for Cassandra. Coordinator is one of the primary components of Stargate which connects the API layer to the backend database. More details can be found [here](https://stargate.io/docs/latest/concepts/concepts.html#stargate-v2-0).
```
Comment (jeffreyscarpenter) [inline README.md]: nit: opensource -> open source

Comment (jeffreyscarpenter) [reply README.md]: and a couple other places

File: README.md (hunk 1 above)
Comment (jeffreyscarpenter) [reply README.md]: Also, should we capitalize Mongoose in text (not commands of course)

File: README.md [hunk 8]
```diff
@@ -1,251 +1,220 @@
...
+| Operation Name      |  Description    |
+| ------------------- | --------------- |
+| Create collection   | `mongoose.model('ModelName',modelSchema)` creates a collection as required |
+| Drop Collection     | `model.dropCollection()` drops the collection |
+| Find Collections    | _Description to be added_ |
+
+### Collection Commands
+| Operation Name      |  Description    |
+| ------------------- | --------------- |
+| Count Documents     | `Model.countDocuments()` returns the count of documents
```
Comment (jeffreyscarpenter) [inline README.md:127]: consistent formatting of operation name column

File: README.md [hunk 9]
```diff
@@ -1,251 +1,220 @@
...
+- JavaScript Clients that use mongoose - mongoose is an elegant mongodb object modeling library for node.js applications. By implementing a driver required by the mongoose interface to connect to the JSON API server instead of native mongodb access layer, now a JavaScript client can store/retrieve documents on an Apache Cassandra/DSE backend.
+
+## Version compatibility
+| Component/Library Name |    Version |
+|------------------------|------------|
+|    mongoose  | 7.x |
+|    Stargate coordinator  | v2.0.9 |
+|    JSON API Server | v1.0.0-ALPHA-2 |
+
+CI tests run using the versions specified in the `api-compatibility.versions` file.
```
Comment (jeffreyscarpenter) [inline README.md]: does this link to actual file?

File: README.md [hunk 10]
```diff
@@ -1,251 +1,220 @@
...
+<img src="stargate-mongoose-arch.png" alt="stargate-mongoose usage end to end architecture"/>
+
+### Components
+- Cassandra Cluster - Apache Cassandra / DataStax Enterprise Cluster as backend database.
+- Stargate Coordinator Nodes - [Stargate](https://stargate.io/) is an opensource Data API Gateway for Cassandra. Coordinator is one of the primary components of Stargate which connects the API layer to the backend database. More details can be found [here](https://stargate.io/docs/latest/concepts/concepts.html#stargate-v2-0).
+- Stargate JSON API - [JSON API Server](https://github.com/stargate/jsonapi) is an opensource JSON API that runs on top of Stargate's coordinator.
+- JavaScript Clients that use mongoose - mongoose is an elegant mongodb object modeling library for node.js applications. By implementing a driver required by the mongoose interface to connect to the JSON API server instead of native mongodb access layer, now a JavaScript client can store/retrieve documents on an Apache Cassandra/DSE backend.
+
+## Version compatibility
+| Component/Library Name |    Version |
```
Comment (jeffreyscarpenter) [inline README.md:97]: won't you have to update this table whenever you update the file?

Maybe the way to do this is to specify the versions of mongoose and DSE (6.8.x) here, and then refer to the file for coordinator and JSON API.

**Review by jeffreyscarpenter — APPROVED** [review]
made several comments but no show stoppers. approving so you are not blocked

File: README.md (hunk 1 above)
Comment (kathirsvn) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (kathirsvn) [reply README.md]: Fixed.

File: README.md (hunk 4 above)
Comment (kathirsvn) [reply README.md:96]: Added a line that says right now, the current implementation uses DSE.

File: README.md (hunk 10 above)
Comment (kathirsvn) [reply README.md:97]: Nice idea, removed JSON API and Stargate from here.

**Review by jeffreyscarpenter — APPROVED** [review]

**Pull request merged**